cmds/wifi/wifi.go

```diff
@@ -44,6 +43,11 @@ func main() {
 		conf  []byte
 	)
 
+	if flag.NArg() < 1 {
```

Comment:
Oh, you're correct. It needs to be called after parse. My bad!

Can you elaborate more on your 2nd and 3rd points?

Reply:
Sure.

Instead of checking the overall number of arguments (maybe there's more in the future) you can specifically check the assigned variable and evaluate the input. For example:

if i == nil || *i == "" {
		panic("i is a required flag")
	}

I'm not suggesting you panic, but you could print the error condition, the usage, and then exit so the user knows what to fix instead of trying to figure out how their usage is different than the expected usage.

If this command is trying to replace some extant userspace utility that _doesn't_ do this, and just prints out a vague error message, then we should probably try to replicate that behavior.

Reply:
Sorry s/i/iface so the previous example becomes

if iface == nil || *iface == "" {
panic("iface is a required flag")
}

Reply:
Actually, you want to do both. The interface should be non-empty *and* there should be 1 arg, looking at the code below.

Reply:
Thanks both for your suggestions!

@erfanian: I agree with what you said about the flag. But for now, since iface is the only flag and it has default value, it will always never be null and it'd be "" only if the user specifically set it to "", so I think our current code should be fine.

@hugelgupf I noticed the current USAGE is as follow: "wifi [options] essid [passphrase]", but in one of the case in the switch statement, we have 3 arguments (the eap case) and it assumes the following format "wifi [options] essid [identity] [passphrase]". So, should we change the format of the USAGE statement?

Also, the original line that cause the panic was this line:
`essid = a[0]`
So by putting the NArg() check, we can handle the no arguments error.  All other errors seem to be handled by the default case of the switch statement.

Reply:
change the usage. Change where you assign essid. YOu don't need the if if you do that.